lexicons/app/bsky/administration/moderationAction.json

```diff
@@ -0,0 +1,21 @@
+{
+  "lexicon": 1,
+  "id": "app.bsky.administration.moderationAction",
+  "defs": {
+    "view": {
+      "type": "object",
+      "required": ["id", "action", "subject", "rationale", "createdBy", "createdAt"],
+      "properties": {
+        "id": {"type": "integer"},
+        "action": {"type": "string", "knownValues": ["takedown"]},
+        "subject": {"type": "union", "refs": ["app.bsky.actor.ref"]},
+        "rationale": {"type": "string"},
+        "createdBy": {"type": "string"},
+        "createdAt": {"type": "string"},
+        "reversedBy": {"type": "string"},
+        "reversedAt": {"type": "string"},
+        "reversedRationale": {"type": "string"}
```

Comment:
I'm the worst. Any way we could put these reversed fields into a subobject that's optional, but then requires all the inner fields?

Reply:
That makes a lot of sense.  What do you think of `reversal: { createdAt, createdBy, reason }`?